### PR TITLE
fix(templates): try/catch handlebars compile

### PR DESCRIPTION
**Pull request by rarkins**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/template/index.ts [hunk 1]
```diff
@@ -228,7 +228,12 @@ export function compile(
       }
     }
   }
-  return handlebars.compile(template)(filteredInput);
+  try {
+    return handlebars.compile(template)(filteredInput);
+  } catch (err) {
+    logger.warn({ err }, 'Error compiling template');
+    return template;
```
Comment (rarkins) [inline lib/util/template/index.ts:235]: Do we want it to return `null`, or the original string?

Comment (viceice) [reply lib/util/template/index.ts:235]: 🤔 should we do a test compile on validate instead? so we catch it at config validation?

Comment (rarkins) [reply lib/util/template/index.ts:235]: I thought we already had something like that. I think the problem I saw in production is caused by release notes content so is seen at runtime only

**Comment by rarkins** [comment]
It seems some users of `template.compile()` rely on it throwing for invalid, so I've made a small change just to satisfy a problem I'm seeing in problem. We can expand safeCompile use over time

File: lib/util/template/index.ts [hunk 2]
```diff
@@ -231,6 +231,19 @@ export function compile(
...
 
+export function safeCompile(
+  template: string,
+  input: CompileInput,
+  filterFields = true
+): string {
+  try {
+    return compile(template, input, filterFields);
+  } catch (err) {
+    logger.warn({ err }, 'Error compiling template');
```
Comment (viceice) [inline lib/util/template/index.ts:242]:
```suggestion
    logger.warn({ err, template }, 'Error compiling template');
```
so we know which template caused the error?

**Review by viceice — APPROVED** [review]

**Pull request merged**